HW15/src/FileLoggerConfiguration.java

```diff
@@ -0,0 +1,83 @@
+import java.io.File;
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.util.Date;
+
+public class FileLoggerConfiguration {
+
+    private static double maxFileSize = 0.3;
+    private static String filePath = "src/";
+    private static File file = createFile();
+    private LoggingLevel loggingLevel;
+    private LogStructure logStructure;
+
+    public FileLoggerConfiguration() {
+    }
+
+    public FileLoggerConfiguration(String filePath, String loggingLevel, double maxFileSize, LogStructure logStructure) {
+        FileLoggerConfiguration.filePath = filePath;
+        this.loggingLevel = LoggingLevel.valueOf(loggingLevel);
+        FileLoggerConfiguration.maxFileSize = maxFileSize;
+        this.logStructure = logStructure;
+    }
+
+    public LogStructure getLogStructure(String message, LoggingLevel loggingLevel) {
+        logStructure = new LogStructure(new Date(), loggingLevel.getLoggingLevel(), message);
+        return logStructure;
+    }
+
+    public boolean isMaxFileSizeExceeded() throws FileMaxSizeReachedException {
+        double currentFileSize = (double) file.length() / 1024;
+        BigDecimal roundedCurrentFileSize = new BigDecimal(currentFileSize).setScale(2, RoundingMode.HALF_UP);
+
+        if (currentFileSize >= maxFileSize) {
```

Comment:
This method, called `isMaxFileSizeExceeded` and returns boolean - so I expect that this method simply returns `true` if the max file size is exceeded and `false` otherwise. And no other additional actions will be done - here new log file will be created - which is not obvious at all.